hax/hax/handler.py

```diff
@@ -232,6 +233,9 @@ def pull_msg():
                     # no op, swallow the exception
                     LOG.exception('**ERROR**')
         except StopIteration:
-            ffi.shun_motr_thread()
+            LOG.info('Consumer Stopped')
         finally:
+            LOG.info('Closing motr')
+            motr.close()
```

Comment:
Previously, Motr context was started in main thread and stopped also there.
With this PR there is a kind of skewness: consumer thread invokes `motr.close()` although it didn't start it. Do we really need this `motr.close()` here in this thread?

Reply:
Yeah, it is observed that during motr finalization, there are messages received from the motr land, thus if all the threads are stopped and motr finalization is invoked from the main thread then the messages seem to block. Hence invoking it from the consumer thread.

Reply:
In this case we probably need to move `m0_addb2_global_thread_leave` at the end of `init_motr_api`. That should help with the `sys->sy_total == 0 ` panic (but I might be wrong here -- I'm assuming init_motr_api is called from the "main" thread and fini_motr_api is called from a non-main thread).

Reply:
You are right, because it addb reference was taken by main thread releasing it in main thread helped. Moved motr_fini to hax.py.
```
May 21 10:24:42 ssc-vm-1623.colo.seagate.com hare-hax[56895]: 2021-05-21 10:24:42,962 [DEBUG] {qconsumer} Shunning Motr thread
May 21 10:24:42 ssc-vm-1623.colo.seagate.com hare-hax[56895]: 2021-05-21 10:24:42,963 [INFO] {qconsumer} Handler thread has exited
May 21 10:24:42 ssc-vm-1623.colo.seagate.com hare-hax[56895]: 2021-05-21 10:24:42,963 [INFO] {MainThread} The http server has stopped
May 21 10:24:42 ssc-vm-1623.colo.seagate.com hare-hax[56895]: 2021-05-21 10:24:42,963 [INFO] {MainThread} Finalizing motr
May 21 10:24:43 ssc-vm-1623.colo.seagate.com hare-hax[56895]: 2021-05-21 10:24:43,034 [DEBUG] {MainThread} Motr API context is down. ByeMay 21 10:24:43 ssc-vm-1623.colo.seagate.com systemd[1]: Stopped HAX daemon for Hare.
```
Finally.